modules/custom/social_scroll/src/SocialScrollManager.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+namespace Drupal\social_scroll;
+
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Drupal\Core\Extension\ModuleHandlerInterface;
+
+/**
+ * Class SocialScrollManager.
+ */
+class SocialScrollManager implements SocialScrollManagerInterface {
+
+  /**
+   * The config factory interface.
+   *
+   * @var \Drupal\Core\Config\ConfigFactoryInterface
+   */
+  protected $configFactory;
+
+  /**
+   * The module handler.
+   *
+   * @var \Drupal\Core\Extension\ModuleHandlerInterface
+   */
+  protected $moduleHandler;
+
+  /**
+   * Constructs a new SocialScrollManager object.
+   *
+   * @param \Drupal\Core\Config\ConfigFactoryInterface $config_factory
+   *   The config factory interface.
+   * @param \Drupal\Core\Extension\ModuleHandlerInterface $module_handler
+   *   The module handler.
+   */
+  public function __construct(ConfigFactoryInterface $config_factory, ModuleHandlerInterface $module_handler) {
+    $this->configFactory = $config_factory;
+    $this->moduleHandler = $module_handler;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getAllAvailableViewIds(): array {
+    return $this->configFactory->getEditable('social_scroll.settings')->getOriginal('views_list') ?? [];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getEnabledViewIds(): array {
+    return array_filter($this->getAllAvailableViewIds());
```

Comment:
I suggest wrapping returned code by the "array_keys" function because actual views identifiers are stored as keys and this method wrapping by this function in all calls.

Reply:
Done.